eZ/Publish/Core/FieldType/User/UserStorage/Gateway/DoctrineStorage.php

```diff
@@ -231,4 +237,176 @@ protected function fetchUserSettings($userId)
+            ->setParameter(':email', $field->value->externalData['email'], ParameterType::STRING)
+            ->setParameter(':passwordHash', $field->value->externalData['passwordHash'], ParameterType::STRING)
+            ->setParameter(':passwordHashType', $field->value->externalData['passwordHashType'], ParameterType::INTEGER)
+            ->where(
+                $queryBuilder->expr()->eq(
+                    $this->connection->quoteIdentifier('contentobject_id'),
+                    ':userId'
+                )
+            )
+            ->setParameter(':userId', $versionInfo->contentInfo->id, ParameterType::INTEGER)
+        ;
+
+        $queryBuilder->execute();
+
+        $settingsQuery = $this->connection->createQueryBuilder();
+
+        $settingsQuery
+            ->update($this->connection->quoteIdentifier(self::USER_SETTING_TABLE))
+            ->set('is_enabled', ':isEnabled')
+            ->set('max_login', ':maxLogin')
+            ->setParameter(':isEnabled', $field->value->externalData['enabled'], ParameterType::INTEGER)
+            ->setParameter(':maxLogin', $field->value->externalData['maxLogin'], ParameterType::INTEGER)
+            ->where(
+                $queryBuilder->expr()->eq(
+                    $this->connection->quoteIdentifier('user_id'),
+                    ':userId'
+                )
+            )
+            ->setParameter(':userId', $versionInfo->contentInfo->id, ParameterType::INTEGER);
+
+        $settingsQuery->execute();
+    }
+
+    public function deleteFieldData(VersionInfo $versionInfo, array $fieldIds): bool
+    {
+        // Delete external storage only, when when deleting last relation to fieldType
+        // to avoid removing it when deleting draft, translation or by exceeding archive limit
+        if (!$this->isLastRelationToFieldType($fieldIds)) {
+            return false;
+        }
+
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete($this->connection->quoteIdentifier(self::USER_SETTING_TABLE))
+            ->where(
+                $query->expr()->eq(
+                    $this->connection->quoteIdentifier('user_id'),
+                    ':userId'
+                )
+            )
+            ->setParameter(':userId', $versionInfo->contentInfo->id, ParameterType::INTEGER);
+
+        $query->execute();
+
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete($this->connection->quoteIdentifier(self::USER_TABLE))
+            ->where(
+                $query->expr()->eq(
+                    $this->connection->quoteIdentifier('contentobject_id'),
+                    ':userId'
+                )
+            )
+            ->setParameter(':userId', $versionInfo->contentInfo->id, ParameterType::INTEGER);
+
+        $query->execute();
+
+        return true;
+    }
+
+    protected function isLastRelationToFieldType(array $fieldIds): bool
+    {
+        foreach ($fieldIds as $fieldId) {
```

Comment:
I'm positive this can be done in one query instead of multiple queries which is always a worse idea.